bin/libast_prereq.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/sh
```

Comment:
this should be bash

Reply:
> this should be bash

Agreed since the script uses features, notably `x=(a b c)` syntax, that is not supported by the real Bourne shell. In practice it probably doesn't matter since most systems make */bin/sh* an alias for */bin/bash* but better to be explicit. Also, there's a lot of weird whitespace in the script.

Reply:
@krader1961: Not sure about your definition of "on most systems", but on Ubuntu machines /bin/sh defaults to **dash** since ~2008 (and probably on most other Debian based boxes since ~2011), on Solaris/SmartOS & friends to [amd64/]ksh93 since ~2010, on older Solaris versions (10-) to a minimal bourne shell! ...

So if non-POSIX features are used, the exact shell name in mind should be used, yes.

Reply:
Argh! Note that on FreeBSD 11 bash lives at */usr/local/bin/bash* and there is no symlink to it from */bin/bash*. We may need to do `#!/usr/bin/env bash` in these scripts. Or perhaps just mandate that either */bin/ksh* or */bin/bash* be valid even if that means the user has to manually create symlinks.

Reply:
While, it's nice to use arrays, shouldn't this just be a plain POSIX script?

Eg.  Instead of:

    iffe_tests_2=( 'tv' 'tmx' )

Just use:

    iffe_tests_2="tv tmx"

Instead of pushd and popd, just use

    cd dir
    cd -

Reply:
Keep in mind, that on some systems /bin/ksh is or points to ksh88 (or even worse pdksh, mksh, ...)!

Reply:
> Keep in mind, that on some systems /bin/ksh is or points to ksh88 (or even worse pdksh, mksh, ...)!

True. But as long as we limit the features we use to the minimum required by POSIX 1003 that shouldn't be a problem. The question is whether we want to have these scripts work with pre-POSIX 1003 shells. I don't think that compatibility is worth it for the same reasons we're no longer going to support pre-ANSI C compilers. We should draw a line in the sand that a POSIX 1003 shell is the bare minimum required to build ksh. Which should not be a problem given that Meson requires a fairly modern Python implementation.

Reply:
This script is just a temporary solution to run ```iffe``` feature tests for Meson builds. I think it would be fine to just depend on bash here. Ideally we should get rid of this script and move all the feature tests to Meson.